operator/pkg/object/objects_test.go

```diff
@@ -712,3 +715,213 @@ func TestK8sObject_ResolveK8sConflict(t *testing.T) {
 		})
 	}
 }
+
+func TestParseK8sObjectsFromYAMLManifestFailOption(t *testing.T) {
+	cases := []struct {
+		name          string
+		manifest      string
+		failOnError   bool
+		expectErr     bool
+		expectCount   int
+		expectedYAMLs []string
+	}{
+		{
+			name: "well formed yaml, no errors",
+			manifest: `
```

Comment:
can you move these lines to single test file per test case?

Reply:
Thanks, will do

Reply:
Fixed